File: Qubit Server/dev/src/org/labrad/qubits/channeldata/Deconvolvable.java
package org.labrad.qubits.channeldata;

import java.util.concurrent.Future;

import org.labrad.qubits.proxies.DeconvolutionProxy;

public interface Deconvolvable {
  /**
   * Whether the data for this block has been deconvolved.
   * @return
   */
  public boolean isDeconvolved();

  /**
   * Mark as needing to be deconvolved again.
   */
  public void invalidate();
  
  /**
   * Deconvolve this item using the provided deconvolver.
   */
  public Future<Void> deconvolve(DeconvolutionProxy deconvolver);
}

File: QubitServer/src/main/scala/org/labrad/qubits/channeldata/Deconvolvable.scala
package org.labrad.qubits.channeldata

import org.labrad.qubits.proxies.DeconvolutionProxy
import scala.concurrent.{ExecutionContext, Future}

trait Deconvolvable {
  /**
   * Whether the data for this block has been deconvolved.
   * @return
   */
  def isDeconvolved(): Boolean

  /**
   * Mark as needing to be deconvolved again.
   */
  def invalidate(): Unit

  /**
   * Deconvolve this item using the provided deconvolver.
   */
  def deconvolve(deconvolver: DeconvolutionProxy)(implicit ec: ExecutionContext): Future[Unit]
}
